packages/flutterfire_cli/lib/src/commands/config.dart

```diff
@@ -186,6 +203,128 @@ class ConfigCommand extends FlutterFireCommand {
     return newProject;
   }
 
+  Future<void> conditionallySetupAndroidGoogleServices({
+    required FlutterApp flutterApp,
+    required FirebaseOptions firebaseOptions,
+    bool force = false,
+  }) async {
+    if (!flutterApp.android) {
+      // Flutter application is not configured to target Android.
+      return;
+    }
+
+    // <app>/android/app/google-services.json
+    var existingProjectId = '';
+    var shouldPromptOverwriteGoogleServicesJson = false;
+    final androidGoogleServicesJsonFile = File(
+      path.join(
+        flutterApp.androidDirectory.path,
+        'app',
+        firebaseOptions.optionsSourceFileName,
+      ),
+    );
+    if (androidGoogleServicesJsonFile.existsSync()) {
+      final existingGoogleServicesJsonContents =
+          await androidGoogleServicesJsonFile.readAsString();
+      existingProjectId = FirebaseAndroidOptions.projectIdFromFileContents(
+        existingGoogleServicesJsonContents,
+      );
+      if (existingProjectId != firebaseOptions.projectId) {
+        shouldPromptOverwriteGoogleServicesJson = true;
+      }
+    }
+    if (shouldPromptOverwriteGoogleServicesJson && !force) {
+      final overwriteGoogleServicesJson = promptBool(
+        'The ${AnsiStyles.cyan(firebaseOptions.optionsSourceFileName)} file already exists but for a different Firebase project (${AnsiStyles.grey(existingProjectId)}). '
+        'Do you want to replace it with new Firebase project ${AnsiStyles.green(firebaseOptions.projectId)}?',
```

Comment:
Not sure on how this reads, around `with new Firebase project` - should it be `with the Firebase project`?